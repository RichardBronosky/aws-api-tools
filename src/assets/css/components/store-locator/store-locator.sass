@import '../../mixins'
@import '../../variables'

.store-locator-page
