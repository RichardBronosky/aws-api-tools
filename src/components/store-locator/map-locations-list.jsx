import React from 'react'
import classNames from 'classnames'
import RTGLink from '../../components/shared/link'
import { letters, getStandardTime, weekdays } from '../../lib/helpers/string-helper.js'
import '../../assets/css/components/store-locator/map-locations-list.sass'

const showDay = (hour, index, currentDay, tomorrowDay) => {
  let daytxt = false
  const day = hour.day ? hour.day : weekdays[hour.dayIndex]
  if (day == currentDay) daytxt = 'Today'
  if (day == tomorrowDay) daytxt = 'Tomorrow'
  if (daytxt) {
    return (
      <div className="hour" key={ index }>
        <span className="day">{ daytxt }</span>: { getStandardTime(hour.openTime) } - { getStandardTime(hour.closeTime) }
      </div>
    )
  }
}

export default ({ markers, onLocationClick, showHours, toggleShowHours, currentDay, tomorrowDay, lia }) => (
  <div className="locations">
    { markers.map((mark, index) => {
      let hours
      if (mark.hours && !mark.hours.regularHours) {
        hours = mark.hours
      } else if (mark.hours && mark.hours.regularHours) {
        hours = mark.hours.regularHours
      }
      const phone = mark.phone || mark.phoneNumber
      return (
        <div className="locationRow" onClick={ () => onLocationClick(index, mark) } key={ index }>
          <div className="imgCol">
            <img src={ `http://maps.google.com/mapfiles/marker${ letters[index] }.png` } />
          </div>
          <div className="addrCol">
            <span className="storeName">
              { !lia ? (
                mark.name
              ) : (
                <RTGLink
                  data={ {
                    url: window && `${ window.location.pathname }?storeCode=${ mark.storeNumber }`,
                    title: `Change Location to ${ mark.name } Store`,
                    category: 'product-lia',
                    action: 'change location click',
                    label: mark.name,
                  } }
                >
                  { mark.name }
                </RTGLink>
              ) }
            </span>
            <br />
            <div className="phone">
              <RTGLink
                data={ {
                  url: `tel:${ phone.replace('/', '-') }`,
                  title: `Call ${ mark.name } Store`,
                  category: 'store-locator',
                  action: 'call store click',
                  label: mark.name,
                } }
              >
                { phone.replace('/', '-') }
              </RTGLink>
            </div>
            <div className="address">
              <RTGLink
                data={ {
                  url: `https://www.google.com/maps/dir/?api=1&origin=Current+Location&destination=${ mark.lat },${ mark.lng }`,
                  title: `Directions to ${ mark.name }`,
                  category: 'store locator',
                  action: 'Directions store map click',
                  label: 'Directions to store',
                } }
                target="_blank"
                className="store-address-link"
              >
                { mark.title }
                <br />
                { mark.city }, { mark.state } { mark.zip }
              </RTGLink>
              <br />
              { mark.distance && <>{ mark.distance } miles</> }
            </div>
            <div className="hours">
              { mark.specialClosings && (
                <span
                  className="special-closings"
                  dangerouslySetInnerHTML={ { __html: mark.specialClosings.markdown.childMarkdownRemark.html } }
                />
              ) }
              <div className="heading">Store Hours</div>
              { hours &&
                hours.map((hour, i) => {
                  if (!showHours[index]) {
                    return showDay(hour, i, currentDay, tomorrowDay)
                  } else {
                    return (
                      <div
                        className={ classNames('hour', {
                          hidden: i > 0 && !showHours[index],
                        }) }
                        key={ i }
                      >
                        <span className="day">{ hour.day ? hour.day : weekdays[hour.dayIndex] }</span>:{ ' ' }
                        { hour.openTime && hour.closeTime
                          ? `${ getStandardTime(hour.openTime) } - ${ getStandardTime(hour.closeTime) }`
                          : 'Closed' }
                      </div>
                    )
                  }
                }) }
            </div>
            <div>
              <button className="showAllBtn" value="Show All" onClick={ () => toggleShowHours(index) }>
                { !showHours[index] ? 'Show All' : 'Show Less' }
              </button>
            </div>
          </div>
        </div>
      )
    }) }
  </div>
)
