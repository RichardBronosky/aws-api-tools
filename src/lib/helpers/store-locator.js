import { weekdays } from './string-helper'
import { getFromBrowserStorage } from './storage'

export const getMarkers = (currentLat, currentLng, storeLocs) => {
  let markers = []
  let stores = storeLocs.filter(store => store.storeTypeId !== 'W')
  for (let i = 0; i < stores.length; i++) {
    stores[i]['distance'] = calculateDistance(
      currentLat,
      currentLng,
      stores[i]['latitude'] || stores[i].location.lat,
      stores[i]['longitude'] || stores[i].location.lon
    )
  }
  stores.sort((a, b) => {
    return a.distance - b.distance
  })
  stores.map((store, index) => {
    const storeDataFromFeed = getStoreInfoFromFeed(store.storeNumber)
    if (storeDataFromFeed && storeDataFromFeed.store_hours && index < 26) {
      markers.push({
        title: store.address1,
        name: getStoreName(store),
        lat: store.location.lat,
        lng: store.location.lon,
        city: store.city,
        state: store.state,
        storeNumber: store.storeNumber,
        zip: store.zip,
        phone: storeDataFromFeed.phone,
        hours: getStoreHours(storeDataFromFeed.store_hours),
        distance: Math.round(store.distance * 10) / 10,
        specialClosings: store.specialClosings,
      })
    }
  })
  return markers
}

export const getStoreName = store => {
  if (store.name && store.storeType) {
    return `${ store.city } - ${ store.name } ${ store.storeType }`
  } else {
    return `${ store.city } ${ store.storeType }`
  }
}

export const getStoreHours = store => {
  let day,
    hours = []
  for (let i = 0, n = weekdays.length; i < n; i++) {
    day = weekdays[i]
    hours.push({
      day: day,
      openTime: store[`${ day.toLowerCase() }Open`],
      closeTime: store[`${ day.toLowerCase() }Closed`],
    })
  }
  return hours
}

export const calculateDistance = (lat1, lon1, lat2, lon2) => {
  if (lat1 == lat2 && lon1 == lon2) {
    return 0
  } else {
    const radlat1 = (Math.PI * lat1) / 180
    const radlat2 = (Math.PI * lat2) / 180
    const theta = lon1 - lon2
    const radtheta = (Math.PI * theta) / 180
    let dist = Math.sin(radlat1) * Math.sin(radlat2) + Math.cos(radlat1) * Math.cos(radlat2) * Math.cos(radtheta)
    if (dist > 1) {
      dist = 1
    }
    return ((Math.acos(dist) * 180) / Math.PI) * 60 * 1.1515 * 0.8684
  }
}

export const getStore = setOrder => {
  const order = setOrder || getFromBrowserStorage('session', 'order')
  if (order && order.distributionIndex) {
    return getStoreInfoFromFeed(order.distributionIndex)
  }
}

export const getStoreInfoFromFeed = storeNum => {
  const storeHoursFeed = require('../generators/data/store-hours-feed.json')
  if (storeNum && storeHoursFeed && storeHoursFeed[storeNum]) {
    return storeHoursFeed[storeNum]
  }
}
