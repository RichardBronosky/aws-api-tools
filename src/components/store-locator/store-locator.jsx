import React from 'react'
import StoreMap from './store-map'
import StoreList from './store-list'
import '../../assets/css/components/store-locator/store-locator.sass'
import { setupAnalytics } from '../../lib/helpers/google-tag-manager.js'

export default class StoreLocator extends React.Component {
  componentDidMount() {
    setupAnalytics({ pageData: { type: 'store', title: 'Store Locator', path: '/stores' } })
  }

  render() {
    const { data } = this.props
    return (
      <div className="store-locator-page">
        <section className="seo-page-heading full-width hide show">
          <div className="grid-container ">
            <div className="grid-x grid-margin-x">
              <div className="cell small-12">
                <h1>Rooms To Go Store Locator</h1>
              </div>
            </div>
          </div>
        </section>
        <StoreMap storeLocations={ data.stores } />
        <StoreList storeLocations={ data.stores } states={ data.states } abbrevStates={ data.abbrevStates } />
      </div>
    )
  }
}
