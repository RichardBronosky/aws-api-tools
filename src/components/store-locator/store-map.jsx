import React from 'react'
import { Map, Marker, InfoWindow, GoogleApiWrapper } from 'google-maps-react'
import { expandState, getLatLng, getCurrentLocation } from '../../lib/helpers/geo-location.js'
import { letters, weekdays, validateZip } from '../../lib/helpers/string-helper.js'
import StoreInfo from './store-info'
import StoreSearchBar from './search-bar'
import MapLocationsList from './map-locations-list'
import { getMarkers } from '../../lib/helpers/store-locator.js'
import '../../assets/css/components/store-locator/store-map.sass'

export class StoreMap extends React.Component {
  state = {
    currentLat: 0,
    currentLng: 0,
    invalidZip: false,
    zipInput: '',
    showHours: [],
    currentDay: '',
    tomorrowDay: '',
    loaded: false,
    activeMarker: {},
    markerObjects: [],
    selectedMarker: {},
    showingInfoWindow: false,
  }

  componentDidMount() {
    const d = new Date()
    let t = new Date()
    t.setDate(d.getDate() + 1)
    this.getInitialCenter(d, t, true)
    this.getInitialCenter(d, t)
  }

  getInitialCenter = (d, t, ip = false) => {
    const location = window.navigator && window.navigator.geolocation
    if (location && !ip) {
      location.getCurrentPosition(
        position => {
          this.setState({
            currentLat: position.coords.latitude,
            currentLng: position.coords.longitude,
            currentDay: weekdays[d.getDay()],
            tomorrowDay: weekdays[t.getDay()],
            loaded: true,
          })
        },
        error => {
          this.updateIPCenter(d, t)
        }
      )
    } else {
      this.updateIPCenter(d, t)
    }
  }

  updateIPCenter = (d, t) => {
    const rtg_location = getCurrentLocation()
    if (rtg_location) {
      this.setState({
        currentLat: rtg_location.lat,
        currentLng: rtg_location.long,
        currentDay: weekdays[d.getDay()],
        tomorrowDay: weekdays[t.getDay()],
        loaded: true,
      })
    }
  }

  updateInputZipValue = evt => {
    this.setState({
      zipInput: evt.target.value,
      invalidZip: false,
    })
  }

  changeZip = async () => {
    if (validateZip(this.state.zipInput)) {
      const location = await getLatLng(this.state.zipInput)
      if (location.lat) {
        this.setState({
          currentLat: location.lat,
          currentLng: location.lng,
        })
      } else {
        this.setState({
          invalidZip: true,
        })
      }
    } else {
      this.setState({
        invalidZip: true,
      })
    }
  }

  toggleShowHours = index => {
    let allShowHours = this.state.showHours
    allShowHours[index] = !this.state.showHours[index]
    this.setState({
      showHours: allShowHours,
    })
  }

  onMarkerMounted = element => {
    if (element && element.marker) {
      this.setState(prevState => {
        return {
          markerObjects: [...prevState.markerObjects, () => element.marker],
        }
      })
    }
  }

  onLocationClick = (index, marker) => {
    const i = parseInt(index)
    if (i > -1 && typeof this.state.markerObjects[index] === 'function') {
      this.setState({
        activeMarker: this.state.markerObjects[index](),
        selectedMarker: marker,
        showingInfoWindow: true,
      })
    }
  }

  onMapClicked = () => {
    if (this.state.showingInfoWindow) {
      this.setState({
        showingInfoWindow: false,
        activeMarker: null,
      })
    }
  }

  render() {
    const { storeLocations } = this.props
    const {
      currentLat,
      currentLng,
      currentDay,
      tomorrowDay,
      showHours,
      activeMarker,
      selectedMarker,
      showingInfoWindow,
      loaded,
      invalidZip,
      zipInput,
    } = this.state
    let abbrevStates = [],
      states = []
    for (let i = 0, n = storeLocations.length; i < n; i++) {
      if (!abbrevStates.includes(storeLocations[i].state)) {
        abbrevStates.push(storeLocations[i].state)
      }
    }
    abbrevStates.sort()
    for (let i = 0, n = abbrevStates.length; i < n; i++) {
      states.push(expandState(abbrevStates[i]))
    }
    const markers = getMarkers(currentLat, currentLng, storeLocations)
    return (
      <div className="store-map grid-x">
        <h2 className="small-12">Store Map</h2>
        <div className="leftScroll">
          <StoreSearchBar
            title="Search For Locations"
            invalidZip={ invalidZip }
            zipInput={ zipInput }
            updateInputZipValue={ this.updateInputZipValue }
            changeZip={ this.changeZip }
          />
          <MapLocationsList
            markers={ markers }
            onLocationClick={ this.onLocationClick }
            showHours={ showHours }
            toggleShowHours={ this.toggleShowHours }
            currentDay={ currentDay }
            tomorrowDay={ tomorrowDay }
          />
        </div>
        { loaded && (
          <Map
            initialCenter={ { lat: currentLat, lng: currentLng } }
            center={ { lat: currentLat, lng: currentLng } }
            className="map"
            google={ this.props.google }
            zoom={ 11 }
            onClick={ this.onMapClicked }
          >
            <Marker
              position={ { lat: currentLat, lng: currentLng } }
              title="Current Location"
              name="Current Location"
              icon={ {
                url: 'http://maps.google.com/mapfiles/marker.png',
              } }
            />
            { markers.map((mark, index) => (
              <Marker
                ref={ this.onMarkerMounted }
                position={ { lat: mark.lat, lng: mark.lng } }
                title={ mark.name }
                icon={ {
                  url: `http://maps.google.com/mapfiles/marker${ letters[index] }.png`,
                } }
                key={ index }
                onClick={ () => this.onLocationClick(index, mark) }
              />
            )) }
            <InfoWindow marker={ activeMarker } visible={ showingInfoWindow }>
              <StoreInfo selectedMarker={ selectedMarker } />
            </InfoWindow>
          </Map>
        ) }
        { !loaded && (
          <div className="map">
            <p className="loading">Loading...</p>
          </div>
        ) }
      </div>
    )
  }
}

export default GoogleApiWrapper({
  apiKey: `${ process.env.GATSBY_GOOGLE_BROWSER_API_KEY }`,
})(StoreMap)
