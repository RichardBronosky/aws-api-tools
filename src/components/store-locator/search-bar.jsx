import React from 'react'
import classNames from 'classnames'
import '../../assets/css/components/store-locator/search-bar.sass'
import { keyboardBlur } from '../../lib/helpers/input-helper'

const onEnterPress = (e, changeZip) => {
  if (e.keyCode == 13 && e.shiftKey == false) {
    changeZip()
    keyboardBlur(e)
  }
}

export default ({ title, invalidZip, zipInput, updateInputZipValue, changeZip }) => {
  return (
    <div className="searchbar">
      <span className="seeStore">{ title }</span>
      <label
        className={ classNames({
          invalid: invalidZip,
        }) }
        htmlFor="location"
      >
        { invalidZip && <p className="invalid-message">*Please enter a valid ZIP code</p> }
        <p className="label-text">Enter ZIP code</p>
        <input
          type="text"
          className="zip-input"
          name="location"
          id="location"
          placeholder="ZIP code"
          aria-label="Enter zip code"
          value={ zipInput }
          onChange={ evt => updateInputZipValue(evt) }
          onKeyDown={ e => onEnterPress(e, changeZip) }
        />
        <button
          value="Change zip code"
          name="change_zip"
          className="submit-btn"
          id="submit-btn"
          aria-label="Change zip code"
          gtm-category="see in store"
          gtm-action="change zip code"
          onClick={ changeZip }
        >
          Submit
        </button>
      </label>
    </div>
  )
}
