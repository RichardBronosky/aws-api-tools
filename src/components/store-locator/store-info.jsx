import React from 'react'
import { Provider } from 'react-redux'
import { store } from '../../redux/store'
import RTGLink from '../shared/link'
import { weekdays, getStandardTime } from '../../lib/helpers/string-helper'
import '../../assets/css/components/store-locator/store-info.sass'

export default ({ selectedMarker }) => {
  let hours
  if (selectedMarker.hours && !selectedMarker.hours.regularHours) {
    hours = selectedMarker.hours
  } else if (selectedMarker.hours && selectedMarker.hours.regularHours) {
    hours = selectedMarker.hours.regularHours
  }
  const currentDay = weekdays[new Date().getDay()]
  return (
    <Provider store={ store }>
      <div className="store-info">
        <h2 className="store-name">{ selectedMarker.name }</h2>
        <div className="storeAddress">
          <RTGLink
            data={ {
              url: `https://www.google.com/maps/dir/?api=1&origin=Current+Location&destination=${ selectedMarker.lat },${ selectedMarker.lng }`,
              title: `Directions to ${ selectedMarker.name }`,
              category: 'store locator',
              action: 'Directions store map click',
              label: 'Directions to store',
            } }
            target="_blank"
            className="store-address-link"
          >
            { selectedMarker.title }
            <br />
            { selectedMarker.city }, { selectedMarker.state } { selectedMarker.zip }
          </RTGLink>
          <br />
          { (selectedMarker.phone || selectedMarker.phoneNumber) && (
            <RTGLink
              data={ {
                url: `tel:${ (selectedMarker.phone ? selectedMarker.phone : selectedMarker.phoneNumber).replace(
                  '/',
                  '-'
                ) }`,
                title: `Call ${ selectedMarker.name }`,
                category: 'store locator',
                action: 'Call store map click',
                label: 'Call store',
              } }
            >
              { (selectedMarker.phone ? selectedMarker.phone : selectedMarker.phoneNumber).replace('/', '-') }
            </RTGLink>
          ) }
          { hours && (
            <div className="today-hours">
              { hours &&
                hours.map((hour, i) => {
                  const day = hour.day ? hour.day : weekdays[hour.dayIndex]
                  if (day === currentDay) {
                    return (
                      <div className="hour" key={ i }>
                        <span className="day">Today</span>: { getStandardTime(hour.openTime) } -{ ' ' }
                        { getStandardTime(hour.closeTime) }
                      </div>
                    )
                  }
                }) }
            </div>
          ) }
        </div>
      </div>
    </Provider>
  )
}
