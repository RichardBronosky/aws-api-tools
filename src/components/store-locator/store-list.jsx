import React from 'react'
import RTGLink from '../shared/link'
import { expandState } from '../../lib/helpers/geo-location.js'
import { slugify } from '../../lib/helpers/string-helper'
import '../../assets/css/components/store-locator/store-list.sass'

export default class StoreList extends React.Component {
  render() {
    const { storeLocations: stores, states, abbrevStates } = this.props
    return (
      <div className="store-list">
        <h2>Store Listings by State</h2>
        <div className="state-list-container">
          { abbrevStates.map((state, index) => (
            <div key={ index } className="state-list">
              <h3>
                <RTGLink
                  data={ {
                    slug: `/stores/${ slugify(states[index]) }`,
                    category: 'store',
                    action: 'click',
                    label: `All stores in ${ states[index] }`,
                  } }
                >
                  { states[index] }
                </RTGLink>
              </h3>
              { stores
                .filter(loc => loc.state === state && loc.storeTypeId !== 'W')
                .map((loc, index) => {
                  let type
                  if (loc.storeTypeId === 'A') {
                    type = 'Showroom'
                  } else if (loc.storeTypeId === 'B') {
                    type = 'EXPRESS'
                  } else if (loc.storeTypeId === 'C') {
                    type = 'Outlet'
                  } else if (loc.storeTypeId === 'K') {
                    type = 'Kids'
                  } else if (loc.storeTypeId === 'P') {
                    type = 'Outdoor'
                  }
                  if (loc.storeName) {
                    return (
                      <div className="link-container" key={ loc.id }>
                        <RTGLink
                          data={ {
                            url: `/stores/${ slugify(expandState(state)).toLowerCase() }/${ slugify(
                              loc.city.toLowerCase()
                            ) }-${ slugify(loc.storeName.toLowerCase()) }-${
                              type ? slugify(type.toLowerCase()) : slugify(loc.storeType.toLowerCase())
                            }-${ loc.storeNumber }`,
                            text: `${ loc.city } - ${ loc.storeName } ${ type || loc.storeType }`,
                            category: 'store',
                            action: 'click',
                            label: `${ loc.city } - ${ loc.storeName } ${ type || loc.storeType }`,
                            value: index,
                          } }
                        />
                      </div>
                    )
                  } else {
                    return (
                      <div className="link-container" key={ loc.id }>
                        <RTGLink
                          data={ {
                            url: `/stores/${ slugify(expandState(state)).toLowerCase() }/${ slugify(
                              loc.city.toLowerCase()
                            ) }-${ type ? slugify(type.toLowerCase()) : slugify(loc.storeType.toLowerCase()) }-${
                              loc.storeNumber
                            }`,
                            text: `${ loc.city } ${ type || loc.storeType }`,
                            category: 'store',
                            action: 'click',
                            label: `${ loc.city } ${ type || loc.storeType }`,
                            value: index,
                          } }
                        />
                      </div>
                    )
                  }
                }) }
            </div>
          )) }
        </div>
      </div>
    )
  }
}
