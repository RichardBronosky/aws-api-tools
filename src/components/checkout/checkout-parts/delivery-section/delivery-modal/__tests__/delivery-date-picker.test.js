import React from 'react'
import renderer from 'react-test-renderer'
import DeliveryDatePicker from '../delivery-date-picker'
import * as storeLocatorHelper from '../../../../../../lib/helpers/store-locator'
import { order1 } from '../../../../../../lib/mocks/checkoutDataMocks'
import { mockStore1 } from '../../../../../../lib/mocks/storeDataMocks'

describe('DeliveryDatePicker', () => {
  const testProps = {
    dates: [
      {
        date: '2019-07-27',
        readable: 'July 27',
        dayOfWeek: 'Sat',
        isPickup: false,
        isStandardDelivery: true,
        isExpressDelivery: false,
      },
    ],
    selectedDate: '2019-07-27',
    setSelectedDate: jest.fn(),
    pickup: false,
    order: order1,
  }

  it('renders correctly when not pickup or express delivery date selected', () => {
    const tree = renderer.create(<DeliveryDatePicker { ...testProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when express delivery date selected', () => {
    const newTestProps = {
      ...testProps,
      dates: [
        {
          date: '2019-07-27',
          readable: 'July 27',
          dayOfWeek: 'Sat',
          isPickup: false,
          isStandardDelivery: true,
          isExpressDelivery: true,
        },
      ],
    }
    const tree = renderer.create(<DeliveryDatePicker { ...newTestProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('renders correctly when pickup date selected', () => {
    storeLocatorHelper.getStore = jest.fn(() => mockStore1)
    const newTestProps = {
      ...testProps,
      dates: [
        {
          date: '2019-07-27',
          readable: 'July 27',
          dayOfWeek: 'Sat',
          isPickup: true,
          isStandardDelivery: true,
          isExpressDelivery: false,
        },
      ],
      pickup: true,
    }
    const tree = renderer.create(<DeliveryDatePicker { ...newTestProps } />).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
