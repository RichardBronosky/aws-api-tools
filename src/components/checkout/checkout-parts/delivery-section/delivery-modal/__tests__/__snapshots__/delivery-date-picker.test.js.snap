// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DeliveryDatePicker renders correctly when express delivery date selected 1`] = `
<div
  className="delivery-date-picker express"
>
  <section
    className="slider"
  >
    <div
      className="slider-head"
    >
      <div
        className="slider-heading"
      >
        
      </div>
      <div
        className="slider-link"
      />
    </div>
  </section>
</div>
`;

exports[`DeliveryDatePicker renders correctly when not pickup or express delivery date selected 1`] = `
<div
  className="delivery-date-picker"
>
  <section
    className="slider"
  >
    <div
      className="slider-head"
    >
      <div
        className="slider-heading"
      >
        
      </div>
      <div
        className="slider-link"
      />
    </div>
  </section>
</div>
`;

exports[`DeliveryDatePicker renders correctly when pickup date selected 1`] = `
<div
  className="delivery-date-picker"
>
  <section
    className="slider"
  >
    <div
      className="slider-head"
    >
      <div
        className="slider-heading"
      >
        
      </div>
      <div
        className="slider-link"
      />
    </div>
  </section>
</div>
`;
